### PR TITLE
Added policy group list on policy group details page

### PR DESCRIPTION
**Pull request by chaitali-mane**
Signed-off-by: chaitali-mane cmane@progress.com

### :nut_and_bolt: Description: What code changed, and why?
We need to add policy files list UI for the policy details page.

### :chains: Related Resources
https://github.com/chef/automate/issues/5344

### :+1: Definition of Done
- Added ```Policyfiles``` tab on the details page 
- Added list on policyfiles for the particular policy group
- Added link on policyfile name, click on policyfile name, open policyfile details page of the same policyfile.

### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab
```
Steps for Policy Groups tab:
Go To Infrastructure -> Chef Servers -> click server name-> click on org name -> Policy Groups tab -> Click on any policy group name -> policy group details page
Where you can see the Policy Groups details page metadata.
Below metadata, you can see a list of policyfiles.

For cypress Build
https://github.com/chef/automate/tree/master/e2e
### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

https://user-images.githubusercontent.com/71449322/129855691-59dcc695-d25b-4e69-8ffb-539a797d3cfb.mp4

### CODE REVIEW

File: components/automate-ui/src/app/modules/infra-proxy/policy-group-details/policy-group-details.component.ts [hunk 1]
```diff
@@ -22,25 +25,39 @@ export class PolicyGroupDetailsComponent implements OnInit, OnDestroy {
   public name: string;
   public show = false;
   public policyCount: number;
+  public policies: Array<any>;
```
Comment (himanshi-chhabra) [inline components/automate-ui/src/app/modules/infra-proxy/policy-group-details/policy-group-details.component.ts:28]: define type for this.

**Review by himanshi-chhabra — APPROVED** [review]

File: components/automate-ui/src/app/modules/infra-proxy/policy-group-details/policy-group-details.component.html [hunk 2]
```diff
@@ -36,7 +36,42 @@ <h3 class="meta-head">Metadata</h3>
...
       </chef-page-header>
+      
+      <section class="page-body" *ngIf="tabValue === 'policyfiles'">
+        <div class="policyfiles-tab" *ngIf="!policyGroupDetailsLoading">
+          <div class="spinner">
+            <chef-loading-spinner *ngIf="policyGroupDetailsLoading" size="50" fixed></chef-loading-spinner>
+          </div>
+          <div data-cy="empty-list" class="empty-section" *ngIf="policyCount === 0">
+            <img alt="No preview" src="/assets/img/no_preview.gif" />
+            <p>No policy files available</p>
```
Comment (tas50) [inline components/automate-ui/src/app/modules/infra-proxy/policy-group-details/policy-group-details.component.html:51]:
```suggestion
            <p>No policyfiles available</p>
```

File: components/automate-ui/src/app/modules/infra-proxy/policy-group-details/policy-group-details.component.html [hunk 3]
```diff
@@ -36,7 +36,42 @@ <h3 class="meta-head">Metadata</h3>
...
+          </div>
+          <div *ngIf="!policyGroupDetailsLoading || policyCount > 0">
+            <chef-table data-cy="policy-group-details-table-container">
+              <chef-thead>
+                <chef-tr class="no_border_tr">
+                  <chef-th class="no_border_th">Policy Files</chef-th>
```
Comment (tas50) [inline components/automate-ui/src/app/modules/infra-proxy/policy-group-details/policy-group-details.component.html:57]:
```suggestion
                  <chef-th class="no_border_th">Policyfiles</chef-th>
```

Comment (chaitali-mane) [reply components/automate-ui/src/app/modules/infra-proxy/policy-group-details/policy-group-details.component.html]: Updated labels, Thanks @tas50 .

**Review by vinay033 — APPROVED** [review]

**Comment by chaitali-mane** [comment]
While acceptance testing found one issue https://github.com/chef/automate/issues/5588

**Pull request merged**